lib/Controller/DevicesApiController.php

```diff
@@ -113,6 +113,10 @@ public function getDevicePoints($id, $pruneBefore=0) {
      */
     public function addDevicePoint($apiversion, $lat, $lng, $timestamp=null, $user_agent=null, $altitude=null, $battery=null, $accuracy=null) {
         if (is_numeric($lat) and is_numeric($lng)) {
+            $timestamp = normalizeOptionalNumber($timestamp);
+            $altitude = normalizeOptionalNumber($altitude);
+            $battery = normalizeOptionalNumber($battery);
+            $accuracy = normalizeOptionalNumber($accuracy);
```

Comment:
For all of these lines:

```suggestion
            $accuracy = $this->normalizeOptionalNumber($accuracy);
```